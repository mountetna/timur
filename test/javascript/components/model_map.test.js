import React from 'react';
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { mockStore } from '../helpers';
import ModelMap from '../../../lib/client/jsx/components/model_map';

const models = {
  monster: { template: require('../fixtures/template_monster.json') },
  labor: { template: require('../fixtures/template_labor.json') },
  project: { template: require('../fixtures/template_project.json') }
};

describe('ModelMap', () => {
  let store;

  it('renders', () => {
    store = mockStore({
      magma: { models },
      janus: { projects: [] }
    });

    global.CONFIG = {
      magma_host: 'magma.test'
    };

    // Wrap with Provider here so store gets passed down to child components in Context
    const tree = renderer
      .create(
        <Provider store={store}>
          <ModelMap />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders an incomplete graph', () => {
    let { monster, project } = models;

    store = mockStore({
      magma: { models: { monster, project } },
      janus: { projects: [] }
    });

    global.CONFIG = {
      magma_host: 'magma.test'
    };

    // Wrap with Provider here so store gets passed down to child components in Context
    const tree = renderer
      .create(
        <Provider store={store}>
          <ModelMap />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
