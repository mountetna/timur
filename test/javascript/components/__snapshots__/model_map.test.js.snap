// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelMap renders 1`] = `
<div
  id="model_map"
>
  <div
    className="map"
  >
    <div
      className="heading report_row"
    >
      <span
        className="name"
      >
        Project
      </span>
       
      <span
        className="title"
      />
    </div>
    <svg
      height={600}
      width={600}
    >
      <defs>
        <marker
          id="arrow"
          markerHeight="3"
          markerUnits="strokeWidth"
          markerWidth="3"
          orient="auto"
          refX="0"
          refY="1"
        >
          <path
            d="M0,0 L0,2 L3,1 z"
          />
        </marker>
      </defs>
      <g
        className="model_link"
      >
        <line
          markerEnd="url(#arrow)"
          x1={300}
          x2={300}
          y1={150}
          y2={280}
        />
      </g>
      <g
        className="model_link"
      >
        <line
          markerEnd="url(#arrow)"
          x1={300}
          x2={300}
          y1={300}
          y2={430}
        />
      </g>
    </svg>
    <div
      className="model_node  "
      onClick={[Function]}
      style={
        Object {
          "left": 300,
          "top": 300,
        }
      }
    >
      labor
    </div>
    <div
      className="model_node  "
      onClick={[Function]}
      style={
        Object {
          "left": 300,
          "top": 450,
        }
      }
    >
      monster
    </div>
    <div
      className="model_node selected "
      onClick={[Function]}
      style={
        Object {
          "left": 300,
          "top": 150,
        }
      }
    >
      project
    </div>
  </div>
  <div
    className="report"
  >
    <div
      className="model_report"
    >
      <div
        className="heading report_row"
      >
        <span
          className="name"
        >
          Model
        </span>
         
        <span
          className="title"
        >
          project
        </span>
      </div>
      <div
        className="heading report_row"
      >
        <span
          className="name"
        >
          Attributes
        </span>
      </div>
      <div
        className="map_attribute report_row"
      >
        <span
          className="type"
        >
           
          identifier
           
        </span>
        <span
          className="value"
        >
          <a
            className="name"
            onClick={[Function]}
          >
            name
          </a>
          <span
            className="description"
          >
            Name for this project
          </span>
        </span>
      </div>
      <div
        className="map_attribute report_row"
      >
        <span
          className="type"
        >
           
          collection
           
        </span>
        <span
          className="value"
        >
          <a
            className="name"
            onClick={[Function]}
          >
            labor
          </a>
        </span>
      </div>
      <div
        className="map_attribute report_row"
      >
        <span
          className="type"
        >
           
          table
           
        </span>
        <span
          className="value"
        >
          <a
            className="name"
            onClick={[Function]}
          >
            codex
          </a>
        </span>
      </div>
    </div>
    <div
      className="attribute_report"
    />
  </div>
</div>
`;

exports[`ModelMap renders an incomplete graph 1`] = `
<div
  id="model_map"
>
  <div
    className="map"
  >
    <div
      className="heading report_row"
    >
      <span
        className="name"
      >
        Project
      </span>
       
      <span
        className="title"
      />
    </div>
    <svg
      height={600}
      width={600}
    >
      <defs>
        <marker
          id="arrow"
          markerHeight="3"
          markerUnits="strokeWidth"
          markerWidth="3"
          orient="auto"
          refX="0"
          refY="1"
        >
          <path
            d="M0,0 L0,2 L3,1 z"
          />
        </marker>
      </defs>
    </svg>
    <div
      className="model_node selected "
      onClick={[Function]}
      style={
        Object {
          "left": 300,
          "top": 300,
        }
      }
    >
      project
    </div>
  </div>
  <div
    className="report"
  >
    <div
      className="model_report"
    >
      <div
        className="heading report_row"
      >
        <span
          className="name"
        >
          Model
        </span>
         
        <span
          className="title"
        >
          project
        </span>
      </div>
      <div
        className="heading report_row"
      >
        <span
          className="name"
        >
          Attributes
        </span>
      </div>
      <div
        className="map_attribute report_row"
      >
        <span
          className="type"
        >
           
          identifier
           
        </span>
        <span
          className="value"
        >
          <a
            className="name"
            onClick={[Function]}
          >
            name
          </a>
          <span
            className="description"
          >
            Name for this project
          </span>
        </span>
      </div>
      <div
        className="map_attribute report_row"
      >
        <span
          className="type"
        >
           
          collection
           
        </span>
        <span
          className="value"
        >
          <a
            className="name"
            onClick={[Function]}
          >
            labor
          </a>
        </span>
      </div>
      <div
        className="map_attribute report_row"
      >
        <span
          className="type"
        >
           
          table
           
        </span>
        <span
          className="value"
        >
          <a
            className="name"
            onClick={[Function]}
          >
            codex
          </a>
        </span>
      </div>
    </div>
    <div
      className="attribute_report"
    />
  </div>
</div>
`;
